Source/AliveLibAO/Abe.cpp

```diff
@@ -2738,11 +2740,393 @@ void Abe::VOn_Tlv_Collision_421130(Path_TLV *pTlv)
     }
 }
 
-void Abe::State_0_Idle_423520()
+void Abe::TryHoist_423420()
 {
     NOT_IMPLEMENTED();
 }
 
+__int16 Abe::HandleDoAction_429A70()
+{
+    NOT_IMPLEMENTED();
+    return 0;
+}
+
+static bool IsSameScaleAsHoist(Path_Hoist* pHoist, BaseAliveGameObject* pObj)
+{
+    auto width = pHoist->field_14_bottom_right.field_0_x - pHoist->field_10_top_left.field_0_x;
+
+    if (pObj->field_BC_sprite_scale == FP_FromInteger(1))
+    {
+        return width > 18;
+    }
+    else
+    {
+        return width <= 18;
+    }
+}
+
+static bool IsFacingSameDirectionAsHoist(Path_Hoist* pHoist, BaseAliveGameObject* pObj)
+{
+    if (pHoist->field_1A_edge_type == Path_Hoist::EdgeType::eLeft && !pObj->field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX))
+    {
+        return false;
+    }
+    else if (pHoist->field_1A_edge_type == Path_Hoist::EdgeType::eRight && pObj->field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX))
+    {
+        return false;
+    }
+    return true;
+}
+
+void Abe::State_0_Idle_423520()
+{
+    Path_TLV *pTlv = nullptr;
+
+    FollowLift_42EE90();
+    if (Input_IsChanting_4334C0() && !field_2A8_flags.Get(Flags_2A8::e2A8_Bit7))
+    {
+        if (field_168_ring_pulse_timer && field_16C_bHaveShrykull)
+        {
+            field_FC_current_motion = eAbeStates::State_162_ToShrykull_42F410;
+            field_110_state = 0;
+        }
+        else
+        {
+            field_114_gnFrame = gnFrameCount_507670 + 90;
+            field_FC_current_motion = eAbeStates::State_150_Chant_42FD50;
+            field_110_state = 0;
+            SND_SEQ_PlaySeq_4775A0(11u, 0, 1);
+        }
+        return;
+    }
+    if (sInputObject_5009E8.isPressed(dword_4C65B8 | dword_4C65DC))
+    {
+        auto held = sInputObject_5009E8.field_0_pads[sCurrentControllerIndex_5076B8].field_6_held;
+        if (held & 0xF0)
+        {
+            field_10C_prev_held = held;
+            field_FC_current_motion = eAbeStates::State_58_ToSpeak_42F8D0;
+            return;
+        }
+    }
+    if (sInputObject_5009E8.isPressed(sInputKey_Hop_4C65A0))
+    {
+        if (sInputObject_5009E8.isPressed(sInputKey_Up_4C6598))
+        {
+            TryHoist_423420();
+        }
+        else if (!sInputObject_5009E8.isPressed(dword_4C65B8 | dword_4C65DC))
+        {
+            field_FC_current_motion = eAbeStates::State_29_HopBegin_4267B0;
+            auto pObj = IntoBirdPortal_402350(2);
```

Comment:
dito

Reply:
It cannot be const, it gets changed potentially.